datafusion/functions/src/string/concat.rs

```diff
@@ -46,7 +45,7 @@ impl ConcatFunc {
     pub fn new() -> Self {
         use DataType::*;
         Self {
-            signature: Signature::variadic(vec![Utf8], Volatility::Immutable),
+            signature: Signature::variadic(vec![Utf8, Utf8View], Volatility::Immutable),
```

Comment:
Perhaps as a follow on PR we can expand this to also support `LargeUtf8`s... though it looks like maybe they are from the tests? Does this just need to be updated and it'll work?

Reply:
I think it's casting to Utf8 from LargeUtf8? Not sure. I can follow up and modify this code to include that though it should be relatively quick.

Reply:
https://github.com/apache/datafusion/pull/12224/files#diff-71970189679c6dd5b3b677bb21603234b488e68d1601be9c4d400d40e430a909R70 pretty sure this line captures the LargeUtf8 usage at the moment.

Reply:
Sorry, should've been clearer, to your point, I think it works but causes a cast to utf8.

```
> CREATE TABLE ttt AS SELECT arrow_cast('T', 'LargeUtf8') AS a;
0 row(s) fetched. 
Elapsed 0.012 seconds.

> SELECt ttt.a FROM ttt;
+---+
| a |
+---+
| T |
+---+
1 row(s) fetched. 
Elapsed 0.007 seconds.

> EXPLAIN SELECt ttt.a FROM ttt;
+---------------+-----------------------------------------------+
| plan_type     | plan                                          |
+---------------+-----------------------------------------------+
| logical_plan  | TableScan: ttt projection=[a]                 |
| physical_plan | MemoryExec: partitions=1, partition_sizes=[1] |
|               |                                               |
+---------------+-----------------------------------------------+
2 row(s) fetched. 
Elapsed 0.007 seconds.

> EXPLAIN SELECt concat(ttt.a, ttt.a) FROM ttt;
+---------------+--------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                       |
+---------------+--------------------------------------------------------------------------------------------+
| logical_plan  | Projection: concat(__common_expr_1 AS ttt.a, __common_expr_1 AS ttt.a)                     |
|               |   Projection: CAST(ttt.a AS Utf8) AS __common_expr_1                                       |
|               |     TableScan: ttt projection=[a]                                                          |
| physical_plan | ProjectionExec: expr=[concat(__common_expr_1@0, __common_expr_1@0) as concat(ttt.a,ttt.a)] |
|               |   ProjectionExec: expr=[CAST(a@0 AS Utf8) as __common_expr_1]                              |
|               |     MemoryExec: partitions=1, partition_sizes=[1]                                          |
|               |                                                                                            |
+---------------+--------------------------------------------------------------------------------------------+
```

Reply:
Correct - I will need to modify the input/return values for this. I'll add an additional test for LargeUtf8 and fix in this PR tonight. It shouldn't take long, I think I have a path forward to change that and it should be relatively painless.

Reply:
Correct** I fixed it now it was really quick resolution lol

Reply:
```
> SELECT ttt.a from ttt;
+---+
| a |
+---+
| T |
+---+
1 row(s) fetched.
Elapsed 0.009 seconds.

> EXPLAIN SELECT ttt.a from ttt;
+---------------+-----------------------------------------------+
| plan_type     | plan                                          |
+---------------+-----------------------------------------------+
| logical_plan  | TableScan: ttt projection=[a]                 |
| physical_plan | MemoryExec: partitions=1, partition_sizes=[1] |
|               |                                               |
+---------------+-----------------------------------------------+
2 row(s) fetched.
Elapsed 0.007 seconds.

> EXPLAIN SELECT concat(ttt.a, ttt.a) from ttt;
+---------------+----------------------------------------------------------------+
| plan_type     | plan                                                           |
+---------------+----------------------------------------------------------------+
| logical_plan  | Projection: concat(ttt.a, ttt.a)                               |
|               |   TableScan: ttt projection=[a]                                |
| physical_plan | ProjectionExec: expr=[concat(a@0, a@0) as concat(ttt.a,ttt.a)] |
|               |   MemoryExec: partitions=1, partition_sizes=[1]                |
|               |                                                                |
+---------------+----------------------------------------------------------------+
2 row(s) fetched.
Elapsed 0.009 seconds.
```

Reply:
Actually it's erroring on cast to generic string array. I need to modify 😆 whoops still not fixed

Reply:
It's resolved 👍